compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -755,15 +755,26 @@ TargetAttr getHIPTargetDetails(StringRef target, StringRef features,
   return nullptr;
 }
 
-Attribute getHIPTargetEncodingLayoutAttr(TargetAttr target) {
+Attribute getHIPTargetEncodingLayoutAttr(TargetAttr target,
+                                         StringRef resolver) {
   // This is only enabled for CDNA2 and CDNA3 for the time being.
   // TODO(kuhar): Enable for other HIP targets.
   if (!llvm::is_contained({"gfx90a", "gfx942"}, target.getArch())) {
     return nullptr;
   }
 
-  return IREE::GPU::GPUPadLayoutAttr::get(
-      target.getContext(), /*cacheLineBytes=*/128, /*cacheSets=*/4);
+  if (resolver == kDataTilingEncodingLayoutResolverName) {
+    // Return a GPUEncodingLayoutAttr with an empty configuration because we
+    // don't want to duplicate the target attribute information in the IR. When
+    // we specialize encodings, the GPUEncodingLayoutAttr will be cloned with
+    // target information just before being resolved into a layout.
```

Comment:
You are exposing Stream context (i.e., encoding specialization) to the codegen. Instead of explaining the current use case, I think we can reword them to something like

```
Return a GPUEncodingLayoutAttr with an empty configuration.
The addtional attributes will be attached by the `cloneWithSimplifiedConfig`
interface method when the resolver needs to be configured.
```